website_sale_affiliate/controllers/main.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl)
+import logging
+
+from odoo.http import request, route
+from odoo.addons.website_sale.controllers.main import WebsiteSale as Base
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteSale(Base):
+    def _store_affiliate_info(self, **kwargs):
```

Comment:
@simahawk - there's a lot of ifs there, vs EAFP which is more Pythonic. 

@hughesbm - Readability would be improved if you reduced nesting. Keep in mind, we want the less nested code as possible in basically all instances.

```
    def _store_affiliate_info(self, **kwargs):
        try:
            request.session['affiliate_id'] = int(kwargs['ref'])
        except KeyError:
            pass
        except ValueError:
            _logger.debug('Invalid affiliate ID value')

        try:
            request.session['affiliate_key'] = kwargs['key']
        except KeyError:
            request.session.pop('affiliate_key', None)
```